src/thor/route_matcher.cc

```diff
@@ -184,12 +202,14 @@ bool expand_from_node(const std::shared_ptr<DynamicCost>* mode_costing,
         prev_edge_label = {kInvalidLabel, edge_id, de, {}, 0, 0, mode, 0};
 
         // Continue walking shape to find the end edge...
-        if (expand_from_node(mode_costing, mode, reader, shape, distances, use_timestamps, index,
-                             end_node_tile, de->endnode(), end_nodes, prev_edge_label, elapsed,
-                             path_infos, false, end_node, followed_edges)) {
+        if (expand_from_node(mode_costing, mode, reader, shape, distances, origin_epoch,
+                             use_timestamps, index, end_node_tile, de->endnode(), end_nodes,
+                             prev_edge_label, elapsed, path_infos, false, end_node, followed_edges)) {
           return true;
         } else {
-          // Match failed along this edge, pop the last entry off path_infos and try the next edge
+          // Match failed along this edge, pop the last entry off path_infos as well as what it
+          // contributed to the elapsed cost/time and try to keep going on the next edge
+          elapsed -= cost;
```

Comment:
so above we add cost on, and here we subtract it back off. logically you can say it would only go negative if floating point round off screwed it up and puts it just below 0 HOWEVER we are inside the node expansion function which means we've already previously added the first edge on the path, ie the elapsed cost was already non zero, ie not close to zero when we entered this function. because of that i dont see how its possible for it to ever go below zero (unless someone writes a costing for which an edge has 0 cost, however this would cause many worse problems before making it to this point)